py/torch_tensorrt/dynamo/lowering/_decompositions.py

```diff
@@ -162,6 +163,45 @@ def var_decomposition(
     return variance
 
 
+@register_torch_trt_decomposition(
+    torch.ops.aten.slice_scatter.default, registry=TORCH_TRT_DECOMPOSITIONS
+)
+def slice_scatter_decomposition(
+    input_tensor: torch.Tensor,
+    src_tensor: torch.Tensor,
+    dim: int,
+    start: Optional[int] = None,
+    end: Optional[int] = None,
+    step: Optional[int] = None,
+):
+    dim_size = input_tensor.shape[dim]
+    start = get_positive_dim(start, input_tensor.shape[dim])
+    if end is None:
+        end = dim_size
+    end = get_positive_dim(end, input_tensor.shape[dim])
+    if step is None:
+        step = 1
+
+    src_dim = src_tensor.shape
+    # step == 0 is not a valid torch case
+    # also src_dim should be equal to slice dimension
+
+    if start == 0 and end == dim_size and step == 1:
+        return src_tensor
+
+    cat_tensors = []
+    index_tensor_shape = []
+    for i, src_each_dim in enumerate(list(src_dim)):
+        if i != dim:
+            index_tensor_shape.append(src_each_dim)
+    for index in range(start, end, step):
+        cat_tensors.append(index * torch.ones(index_tensor_shape))
+    index_tensor = torch.stack(cat_tensors, dim)
+    index_tensor = index_tensor.to(torch.int64).cuda()
+    output_tensor = torch.scatter(input_tensor, dim, index_tensor, src)
+    return output_tensor
```

Comment:
Could this potentially be simplified to avoid `for`-loops using `torch.arange`? For instance, see this [implementation](https://github.com/pytorch/pytorch/blob/e5e0685f614171ab3e3a22c85dd4ee8cd0cf09ae/aten/src/ATen/functorch/BatchRulesScatterOps.cpp#L872-L879)

Reply:
Hi @gs-olive I tried the above implementation.
I am not sure how `get_expanded_index` works, but I think it will be difficult to achieve the above behavior without for loops.
I tried two alternate
```
indices = torch.arange(start,stop, step)
cat_tensors = torch.unsqueeze(indices,1) * torch.ones(index_tensor_shape)).flatten(end_dim=1)
#or
cat_tensors = indices(:, None) * torch.ones(index_tensor_shape)).flatten(end_dim=1)
```
The thing is we need to unsqueeze indices n no of times, where n is the dimension of index_tensor_shape. While the above would work for cases
```
input = torch.ones(8,8)
src = torch.ones(8,2)
out  = torch.slice_scatter(input, src, 1, 6, 8, 1) 
```
or 
```
input = torch.ones(8,8)
src = torch.ones(8,1)
out  = torch.slice_scatter(input, src, 1, 6, 7, 1) 
```
it would start failing for input and src with sizes `torch.ones(8,8,8)` and `torch.zeros(8,2,8)` or `torch.zeros(8,1,8) `respectively. We would have to unsqueeze n no of times, eg: torch.unsqueeze(indices,1,1) or indices[:,None,None] would work, but then that would again be a for loop.

I cannot think of another way on top of my mind, if you have any suggestion you could let me know. 
For now the test cases pass with for loop so I have reverted back to that,